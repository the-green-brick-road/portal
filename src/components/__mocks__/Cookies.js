/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }             from '@mui/material';

function Cookies(props) {

    /* --------- Gather inputs --------- */
    const { color } = props;

    /* -------- Defining theme --------- */
    return (
        <Box style={{ minHeight:'30px',zIndex:1000, position:'absolute', backgroundColor:color }} >
            <Typography> Cookies </Typography>
        </Box>
    );

}

export default Cookies;
