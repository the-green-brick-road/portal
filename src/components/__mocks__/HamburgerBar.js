/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# HamburgerBar component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }   from '@mui/material';

function HamburgerBar(props) {

    /* --------- Gather inputs --------- */
    const { height, isNegative, theme={} }  = props
    /* const componentName = 'HamburgerBar'; */

    /* -------- Defining theme --------- */
    let color = theme.palette.primary.main;
    if (isNegative) { color = theme.palette.common.white; }

    return (
        <Box style={{height:height, backgroundColor:color, width: '100%'}}>
            <Typography> HamburgerBar </Typography>
        </Box>
    );

}

export default HamburgerBar;
