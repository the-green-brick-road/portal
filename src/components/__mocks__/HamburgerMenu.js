/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# HamburgerMenu component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box, Typography }   from '@mui/material';

function HamburgerMenu(props) {

    /* --------- Gather inputs --------- */
    const { top , margin, isNegative, theme={} } = props;
    /* const componentName = 'HamburgerBar'; */

    /* -------- Defining theme --------- */
    let color = theme.palette.primary.main;
    if (isNegative) { color = theme.palette.common.white; }
    const menuWidthString = `calc( 100vw - 2 * ${margin})`;

    return (
        <Box style={{ minHeight:'30px', width:menuWidthString, top:top, zIndex:1000, position:'absolute', backgroundColor:color }} >
            <Typography> HamburgerMenu </Typography>
        </Box>
    );

}

export default HamburgerMenu;