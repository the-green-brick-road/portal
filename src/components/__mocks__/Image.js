/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @24 may 2023
# Latest revision: 24 may 2023
# ---------------------------------------------------- */

/* Material UI includes */
import { Box }             from '@mui/material';

function Image(props) {

    /* --------- Gather inputs --------- */
    const { style={} }    = props;

    const local_style = JSON.parse(JSON.stringify(style))

    if (!('backgroundColor' in local_style)) { local_style['backgroundColor'] = 'blue' }

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Box style={local_style} />
    );
    /* eslint-enable padded-blocks */

}

export default Image;
