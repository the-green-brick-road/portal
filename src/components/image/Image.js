/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Image component
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @22 may 2023
# Latest revision: 14 june 2023
# ---------------------------------------------------- */

/* React includes */
import { Profiler }              from 'react';

/* Portal includes */
import { useDesign, useLogging } from '../../providers';

function Image(props) {

    /* --------- Gather inputs --------- */
    const { name, style={} }    = props;
    const { images }            = useDesign();
    const { onRender }          = useLogging();
    const componentName = 'Image';

    /* ---- Select image if loaded ----- */
    let data = {}
    if (name in images) { data = images[name] }
    const label_picture = `picture-${name}`

    /* ----------- Define HTML --------- */
    /* eslint-disable padded-blocks */
    return (
        <Profiler id={componentName} onRender={onRender}>
            {(data !== {}) && (
                <picture style={style} data-testid={label_picture}>
                    { Object.entries(data).toReversed().map((item) => {

                        let result;
                        const localType = `image/${item[0]}`;
                        const localSizes = Object.entries(item[1]);
                        let localSrcset = '';
                        for (let i = 0; i < localSizes.length; i += 1) {
                            localSrcset = `${localSrcset} ${localSizes[i][1].img} ${localSizes[i][1].width} ,`;
                        }
                        if (item[0] !== 'default') {
                            result = (<source key={item[0]} srcSet={localSrcset} type={localType} />);
                        }
                        return result;
                    })}
                    <img src={data.default} style={style} alt={name}/>
                </picture>
            )}
        </Profiler>
    );
    /* eslint-enable padded-blocks */

}

export default Image;
