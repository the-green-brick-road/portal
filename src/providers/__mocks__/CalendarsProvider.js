/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Calendars provider mock
# -------------------------------------------------------
# Nad√®ge LEMPERIERE, @14 june 2023
# Latest revision: 14 june 2023
# ---------------------------------------------------- */

/* React includes */
import { createContext }       from 'react';

const context = createContext(null);

/* ----- Define provider values ---- */
var state = { open: {} };

function CalendarsProvider(props) {

    const { children, open={} } = props;

    /* ------ Defining behaviour ------- */
    state.open = open;

    /* ----------- Define HTML --------- */
    return (
        <context.Provider value={state}>
            {children}
        </context.Provider>
    );

}

function useCalendars() { return state; };

export { useCalendars, CalendarsProvider };