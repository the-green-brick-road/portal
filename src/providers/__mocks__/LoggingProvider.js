/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Logging provider mock
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @09 may 2023
# Latest revision: 09 may 2023
# ---------------------------------------------------- */

/* React includes */
import { createContext }       from 'react';

const context = createContext(null);

/* ----- Define provider values ---- */
var state = {
    isLoggingActivated : true,
    logText: jest.fn((component, level, topic, message) => {  console.log(`${component} - ${level} - ${topic} -${message}`) }),
};

function LoggingProvider(props) {

    const { children } = props;

    /* ----------- Define HTML --------- */
    return (
        <context.Provider value={state}>
            {children}
        </context.Provider>
    );

}

function useLogging() { return state; };

export { useLogging, LoggingProvider };