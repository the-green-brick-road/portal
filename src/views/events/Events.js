/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Events view component
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @11 june 2023
# Latest revision: 12 june 2023
# ---------------------------------------------------- */

/* React includes */
import { Fragment, useState, useEffect }             from 'react';

/* Material UI includes */
import { Typography, Container, Table, TableBody }   from '@mui/material';
import { TableRow, TableCell, Chip, Stack }          from '@mui/material';
import { default as LocationOn }                     from '@mui/icons-material/LocationOn';
import { useTheme }                                  from '@mui/material/styles';

/* Portal includes */
import { useData }                                   from '../../providers';
import { Image }                                     from '../../components';

/* Local includes */
import { default as EventsDivider }                  from './EventsDivider';

function Competitions() {

    /* --------- Gather inputs --------- */
    const theme                           = useTheme();
    const { calendars }                   = useData();
    const [ localEvents, setLocalEvents ] = useState([])
    const months   = ["Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec"]
    const days   = ["Sun", "Mon", "Tue", "Wed", "Thu", "Fri", "Sat"]

    /* ------- Preprocess competitions ------- */
    useEffect(() => {

        const now = new Date()
        const month = now.getMonth();
        const year = now.getYear() + 1900;
        const date = now.getDate();
        const day = now.getDay()
        const local_events = {}
        local_events[year] = {}
        local_events[year][month] = {}
        local_events[year][month][date] = {
            day:day,
            now:true,
            events:[
            ],
        }

        if ('public' in calendars) {

            for (let i_event = 0; i_event < calendars.public.length; i_event +=1) {

                let date_start = Date()
                let date_end = Date()
                if ( 'date' in calendars.public[i_event].start ) {

                    date_start = new Date(calendars.public[i_event].start.date)
                    const offset = date_start.getTimezoneOffset()
                    date_start = new Date(date_start.getTime() + offset * 60 * 1000)

                }
                else if ( 'dateTime' in calendars.public[i_event].start ) {

                    const temp = new Date(Date.parse(calendars.public[i_event].start.dateTime))
                    temp.toLocaleString("en-US", {timeZone: calendars.public[i_event].start.timeZone})
                    date_start = temp

                }
                if ( 'date' in calendars.public[i_event].end )   {

                    date_end = new Date(calendars.public[i_event].end.date)
                    const offset = date_end.getTimezoneOffset()
                    date_end = new Date(date_end.getTime() + offset * 60 * 1000 - 1)

                }
                else if ( 'dateTime' in calendars.public[i_event].end )   {

                    const temp = new Date(Date.parse(calendars.public[i_event].end.dateTime))
                    temp.toLocaleString("en-US", {timeZone: calendars.public[i_event].end.timeZone})
                    date_end = temp

                }

                const ms = date_start.getMonth();
                const ys = date_start.getYear() + 1900;
                const dts = date_start.getDate();
                const das = date_start.getDay();
                const me = date_end.getMonth();
                const ye = date_end.getYear() + 1900;
                const dte = date_end.getDate();
                const dae = date_end.getDay();

                const shall_add_end = (ys !== ye) || (ms !== me) || (dts !== dte)

                if (!( ys in local_events)) { local_events[ys] = {}}
                if (!( ms in local_events[ys])) { local_events[ys][ms] = {}}
                if (!( dts in local_events[ys][ms])) { local_events[ys][ms][dts] = {day:das, now:false, events:[]} }
                if (!shall_add_end) {

                    local_events[ys][ms][dts].events.push({
                        start:date_start.toLocaleTimeString('en-US', {hour: '2-digit', minute:'2-digit'}),
                        end:date_end.toLocaleTimeString('en-US', {hour: '2-digit', minute:'2-digit'}),
                        type:'public',
                        summary:calendars.public[i_event].summary,
                        location:calendars.public[i_event].location,
                    })

                }
                else {

                    local_events[ys][ms][dts].events.push(
                        {
                            start:date_start.toLocaleTimeString('en-US', {hour: '2-digit', minute:'2-digit'}),
                            type:'public',
                            summary:calendars.public[i_event].summary,
                            location:calendars.public[i_event].location,
                        }

                    )
                    if (!( ye in local_events)) { local_events[ye] = {}}
                    if (!( me in local_events[ys])) { local_events[ye][me] = {}}
                    if (!( dte in local_events[ys][ms])) { local_events[ye][me][dte] = {day:dae, now:false, events:[]} }
                    local_events[ye][me][dte].events.push(
                        {
                            end:date_end.toLocaleTimeString('en-US', {hour: '2-digit', minute:'2-digit'}),
                            type:'public',
                            summary:calendars.public[i_event].summary,
                            location:calendars.public[i_event].location,
                        }

                    )

                }

            }

        }


        setLocalEvents(local_events)

    }, [calendars]);

    /* ----------- Define HTML --------- */
    return (
        <Fragment>
            <Container style={{ width:'100%', padding:0, position:'relative' }}>
                <Image name="events" style={{ width:'100%', pointerEvents: 'none' }}/>
            </Container>
            <Container>
                <Table>
                    <TableBody>
                        { Object.entries(localEvents).map((item, index) => { /* Loop on all members to create entries */

                            return(
                                <Fragment key={item}>
                                    { Object.entries(item[1]).map((jtem, jndex) => { /* Loop on all members to create entries */

                                        const label = `${months[jtem[0]]} ${item[0]}`
                                        return(
                                            <Fragment key={jtem}>
                                                <TableRow key={jndex}>
                                                    <TableCell colSpan={4} style={{border:'none'}}>
                                                        <EventsDivider textAlign='left'col={theme.palette.primary.main} >
                                                            <Chip style={{ color:theme.palette.primary.main }} label={label} />
                                                        </EventsDivider>
                                                    </TableCell>
                                                </TableRow>
                                                { Object.entries(jtem[1]).map((ktem, kndex) => { /* Loop on all members to create entries */

                                                    return(
                                                        <TableRow key={kndex}>
                                                            {(ktem[1].now) && (
                                                                <TableCell colSpan={1} style={{backgroundColor:theme.palette.primary.main, padding:5}}>
                                                                    <Typography style={{ textAlign:'center', color:theme.palette.common.white }}> {days[ktem[1].day]} </Typography>
                                                                    <Typography style={{ textAlign:'center', color:theme.palette.common.white }}> {[ktem[0]]} </Typography>
                                                                </TableCell>
                                                            )}
                                                            {!(ktem[1].now) && (
                                                                <TableCell colSpan={1} style={{padding:5}}>
                                                                    <Typography style={{ textAlign:'center' }}> {days[ktem[1].day]} </Typography>
                                                                    <Typography style={{ textAlign:'center' }}> {[ktem[0]]} </Typography>
                                                                </TableCell>
                                                            )}
                                                            <TableCell colSpan={3} style={{padding:5}}>
                                                                {(ktem[1].now) && (<Typography style={{ color:theme.palette.primary.main, fontSize:'14px'}}> Today </Typography>)}
                                                                { ktem[1].events.map((ltem, lndex) => { /* Loop on all members to create entries */

                                                                    return (
                                                                        <Fragment key={lndex}>
                                                                            <Typography style={{fontSize:'14px'}}>
                                                                                {('start' in ltem) && ( ltem.start )}
                                                                                {('start' in ltem) && ('end' in ltem) && ( ' - ' )}
                                                                                {('end' in ltem) && ( ltem.end )}
                                                                                : {ltem.summary}
                                                                                {('start' in ltem) && (!('end' in ltem)) && ( ' [START]' )}
                                                                                {(!('start' in ltem)) && ('end' in ltem) && ( ' [END]' )}
                                                                            </Typography>
                                                                            <Stack direction="row" alignItems="center">
                                                                                <LocationOn style={{height:'14px'}}/>
                                                                                <Typography style={{fontSize:'11px'}}> {ltem.location} </Typography>
                                                                            </Stack>
                                                                        </Fragment>
                                                                    )

                                                                })}

                                                            </TableCell>
                                                        </TableRow>
                                                    )

                                                })}

                                            </Fragment>
                                        )

                                    })}
                                </Fragment>
                            )


                        })}
                    </TableBody>
                </Table>
            </Container>
        </Fragment>
    );

}

export default Competitions;