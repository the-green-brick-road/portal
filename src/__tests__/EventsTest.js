/* ------------------------------------------------------
# THE GREEN BRICK ROAD
# -------------------------------------------------------
# Copyright (c) [2023] The Green Brick Road
# All rights reserved
# -------------------------------------------------------
# Events view component test suite
# -------------------------------------------------------
# NadÃ¨ge LEMPERIERE, @30 may 2023
# Latest revision: 30 may 2023
# ---------------------------------------------------- */

/* Test includes */
import { render, prettyDOM }       from '@testing-library/react'
import { expect, test}             from '@jest/globals';

/* Component under test */
import { default as Events }       from '../views/events/Events';

/* Mocks includes */
/* eslint-disable jest/no-mocks-import */
import { default as MockImage }    from '../components/__mocks__/Image';
import { useCalendars as mockUseCalendars, CalendarsProvider as MockCalendarsProvider } from '../providers/__mocks__/CalendarsProvider';
import { useLogging as mockUseLogging, LoggingProvider as MockLoggingProvider } from '../providers/__mocks__/LoggingProvider';
jest.mock("../components", () => ({ Image: (props) => MockImage(props) }));
jest.mock('../providers', () => ({
    useCalendars: (() => { return mockUseCalendars(); }),
    useLogging: (() => { return mockUseLogging() }),
}));
/* eslint-enable jest/no-mocks-import */

describe("Events view" ,() => {

    test('Should display events page', async () => {

        const Open = [

            {'summary' : 'event1', 'location' : 'location1', start:{date : '2013-05-17'}, end : {dateTime:'2013-05-19T16:00:00-05:00', timeZone: 'America/New_York'}},
            {'summary' : 'event2', 'location' : 'location2', start:{date : '2012-05-16'}, end : {dateTime:'2012-05-16T16:00:00-05:00', timeZone: 'America/New_York'}},
            {'summary' : 'event3', 'location' : 'location3', start:{date : '2013-05-20'}, end : {date : '2013-05-20'}}

        ];

        const view = render(
            <div>
                <MockLoggingProvider>
                    <MockCalendarsProvider open={Open}>
                        <Events/>
                    </MockCalendarsProvider>
                </MockLoggingProvider>
            </div>
        );

        /*Remove current date display since it changes between each test */
        document.getElementById('now').remove(); /* eslint-disable-line testing-library/no-node-access */

        const tree = prettyDOM(view.baseElement, Number.POSITIVE_INFINITY, {filterNode: () => true, escapeString: false, highlight: false});
        expect(tree).toMatchSnapshot();

    })

})
