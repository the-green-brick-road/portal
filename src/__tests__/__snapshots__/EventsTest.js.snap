// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Events view Should display events page 1`] = `
"<body>
  <div
    id=\\"root\\"
  />
  <div>
    <div>
      <div
        class=\\"MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root\\"
        style=\\"width: 100%; padding: 0px; position: relative;\\"
      >
        <div
          class=\\"MuiBox-root css-0\\"
          style=\\"width: 100%; pointer-events: none; height: auto; background-color: blue;\\"
        />
      </div>
      <div
        class=\\"MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root\\"
      >
        <table
          class=\\"MuiTable-root css-rqglhn-MuiTable-root\\"
        >
          <tbody
            class=\\"MuiTableBody-root css-apqrd9-MuiTableBody-root\\"
          >
            <tr
              class=\\"MuiTableRow-root css-1q1u3t4-MuiTableRow-root\\"
            >
              <td
                class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root\\"
                colspan=\\"4\\"
              >
                <div
                  class=\\"MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren MuiDivider-textAlignLeft css-yt76fy-MuiDivider-root\\"
                  col=\\"#1976d2\\"
                  role=\\"separator\\"
                >
                  <span
                    class=\\"MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper\\"
                  >
                    <div
                      class=\\"MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gavykb-MuiChip-root\\"
                      style=\\"color: rgb(25, 118, 210);\\"
                    >
                      <span
                        class=\\"MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label\\"
                      >
                        May 2012
                      </span>
                    </div>
                  </span>
                </div>
              </td>
            </tr>
            <tr
              class=\\"MuiTableRow-root css-1q1u3t4-MuiTableRow-root\\"
            >
              <td
                class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root\\"
                colspan=\\"1\\"
                style=\\"padding: 5px;\\"
              >
                <p
                  class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                  style=\\"text-align: center;\\"
                >
                   
                  Wed
                   
                </p>
                <p
                  class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                  style=\\"text-align: center;\\"
                >
                   
                  16
                   
                </p>
              </td>
              <td
                class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root\\"
                colspan=\\"3\\"
                style=\\"padding: 5px;\\"
              >
                <p
                  class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                  style=\\"font-size: 14px;\\"
                >
                  12:00 AM
                   - 
                  09:00 PM
                  : 
                  event2
                </p>
                <div
                  class=\\"MuiStack-root css-1d9cypr-MuiStack-root\\"
                >
                  <svg
                    aria-hidden=\\"true\\"
                    class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
                    data-testid=\\"LocationOnIcon\\"
                    focusable=\\"false\\"
                    style=\\"height: 14px;\\"
                    viewBox=\\"0 0 24 24\\"
                  >
                    <path
                      d=\\"M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7m0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5\\"
                    />
                  </svg>
                  <p
                    class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                    style=\\"font-size: 11px;\\"
                  >
                     
                    location2
                     
                  </p>
                </div>
              </td>
            </tr>
            <tr
              class=\\"MuiTableRow-root css-1q1u3t4-MuiTableRow-root\\"
            >
              <td
                class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root\\"
                colspan=\\"4\\"
              >
                <div
                  class=\\"MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren MuiDivider-textAlignLeft css-yt76fy-MuiDivider-root\\"
                  col=\\"#1976d2\\"
                  role=\\"separator\\"
                >
                  <span
                    class=\\"MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper\\"
                  >
                    <div
                      class=\\"MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gavykb-MuiChip-root\\"
                      style=\\"color: rgb(25, 118, 210);\\"
                    >
                      <span
                        class=\\"MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label\\"
                      >
                        May 2013
                      </span>
                    </div>
                  </span>
                </div>
              </td>
            </tr>
            <tr
              class=\\"MuiTableRow-root css-1q1u3t4-MuiTableRow-root\\"
            >
              <td
                class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root\\"
                colspan=\\"1\\"
                style=\\"padding: 5px;\\"
              >
                <p
                  class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                  style=\\"text-align: center;\\"
                >
                   
                  Fri
                   
                </p>
                <p
                  class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                  style=\\"text-align: center;\\"
                >
                   
                  17
                   
                </p>
              </td>
              <td
                class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root\\"
                colspan=\\"3\\"
                style=\\"padding: 5px;\\"
              >
                <p
                  class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                  style=\\"font-size: 14px;\\"
                >
                  12:00 AM
                  : 
                  event1
                   [START]
                </p>
                <div
                  class=\\"MuiStack-root css-1d9cypr-MuiStack-root\\"
                >
                  <svg
                    aria-hidden=\\"true\\"
                    class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
                    data-testid=\\"LocationOnIcon\\"
                    focusable=\\"false\\"
                    style=\\"height: 14px;\\"
                    viewBox=\\"0 0 24 24\\"
                  >
                    <path
                      d=\\"M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7m0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5\\"
                    />
                  </svg>
                  <p
                    class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                    style=\\"font-size: 11px;\\"
                  >
                     
                    location1
                     
                  </p>
                </div>
              </td>
            </tr>
            <tr
              class=\\"MuiTableRow-root css-1q1u3t4-MuiTableRow-root\\"
            >
              <td
                class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root\\"
                colspan=\\"1\\"
                style=\\"padding: 5px;\\"
              >
                <p
                  class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                  style=\\"text-align: center;\\"
                >
                   
                  Sun
                   
                </p>
                <p
                  class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                  style=\\"text-align: center;\\"
                >
                   
                  19
                   
                </p>
              </td>
              <td
                class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root\\"
                colspan=\\"3\\"
                style=\\"padding: 5px;\\"
              >
                <p
                  class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                  style=\\"font-size: 14px;\\"
                >
                  09:00 PM
                  : 
                  event1
                   [END]
                </p>
                <div
                  class=\\"MuiStack-root css-1d9cypr-MuiStack-root\\"
                >
                  <svg
                    aria-hidden=\\"true\\"
                    class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
                    data-testid=\\"LocationOnIcon\\"
                    focusable=\\"false\\"
                    style=\\"height: 14px;\\"
                    viewBox=\\"0 0 24 24\\"
                  >
                    <path
                      d=\\"M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7m0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5\\"
                    />
                  </svg>
                  <p
                    class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                    style=\\"font-size: 11px;\\"
                  >
                     
                    location1
                     
                  </p>
                </div>
                <p
                  class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                  style=\\"font-size: 14px;\\"
                >
                  11:59 PM
                  : 
                  event3
                   [END]
                </p>
                <div
                  class=\\"MuiStack-root css-1d9cypr-MuiStack-root\\"
                >
                  <svg
                    aria-hidden=\\"true\\"
                    class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
                    data-testid=\\"LocationOnIcon\\"
                    focusable=\\"false\\"
                    style=\\"height: 14px;\\"
                    viewBox=\\"0 0 24 24\\"
                  >
                    <path
                      d=\\"M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7m0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5\\"
                    />
                  </svg>
                  <p
                    class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                    style=\\"font-size: 11px;\\"
                  >
                     
                    location3
                     
                  </p>
                </div>
              </td>
            </tr>
            <tr
              class=\\"MuiTableRow-root css-1q1u3t4-MuiTableRow-root\\"
            >
              <td
                class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root\\"
                colspan=\\"1\\"
                style=\\"padding: 5px;\\"
              >
                <p
                  class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                  style=\\"text-align: center;\\"
                >
                   
                  Mon
                   
                </p>
                <p
                  class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                  style=\\"text-align: center;\\"
                >
                   
                  20
                   
                </p>
              </td>
              <td
                class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root\\"
                colspan=\\"3\\"
                style=\\"padding: 5px;\\"
              >
                <p
                  class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                  style=\\"font-size: 14px;\\"
                >
                  12:00 AM
                  : 
                  event3
                   [START]
                </p>
                <div
                  class=\\"MuiStack-root css-1d9cypr-MuiStack-root\\"
                >
                  <svg
                    aria-hidden=\\"true\\"
                    class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\"
                    data-testid=\\"LocationOnIcon\\"
                    focusable=\\"false\\"
                    style=\\"height: 14px;\\"
                    viewBox=\\"0 0 24 24\\"
                  >
                    <path
                      d=\\"M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7m0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5\\"
                    />
                  </svg>
                  <p
                    class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                    style=\\"font-size: 11px;\\"
                  >
                     
                    location3
                     
                  </p>
                </div>
              </td>
            </tr>
            <tr
              class=\\"MuiTableRow-root css-1q1u3t4-MuiTableRow-root\\"
            >
              <td
                class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root\\"
                colspan=\\"4\\"
              >
                <div
                  class=\\"MuiDivider-root MuiDivider-fullWidth MuiDivider-withChildren MuiDivider-textAlignLeft css-yt76fy-MuiDivider-root\\"
                  col=\\"#1976d2\\"
                  role=\\"separator\\"
                >
                  <span
                    class=\\"MuiDivider-wrapper css-qywfm8-MuiDivider-wrapper\\"
                  >
                    <div
                      class=\\"MuiChip-root MuiChip-filled MuiChip-sizeMedium MuiChip-colorDefault MuiChip-filledDefault css-gavykb-MuiChip-root\\"
                      style=\\"color: rgb(25, 118, 210);\\"
                    >
                      <span
                        class=\\"MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label\\"
                      >
                        Feb 2024
                      </span>
                    </div>
                  </span>
                </div>
              </td>
            </tr>
            <tr
              class=\\"MuiTableRow-root css-1q1u3t4-MuiTableRow-root\\"
            >
              <td
                class=\\"MuiTableCell-root MuiTableCell-body MuiTableCell-sizeMedium css-1ex1afd-MuiTableCell-root\\"
                colspan=\\"3\\"
                style=\\"padding: 5px;\\"
              >
                <p
                  class=\\"MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root\\"
                  style=\\"color: rgb(25, 118, 210); font-size: 14px;\\"
                >
                   Today 
                </p>
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </div>
</body>"
`;
